drake/solvers/NonlinearProgramSnoptmex.cpp

```diff
@@ -60,6 +69,16 @@ void mexFunction(int nlhs, mxArray* plhs[], int nrhs, const mxArray* prhs[])
   {
     mexErrMsgIdAndTxt("Drake:NonlinearProgramSnoptmex:InvalidInputs","The usage is [x,F,info,xmul,Fmul] = NonlinearProgramSnoptmex(x,xlow,xupp,Flow,Fupp,userfun,ObjAdd,ObjRow,A,iAfun,jAvar,iGfun,jGvar,options)");
   }
+
+  if (lenrw==0) { // then initialize (sninit needs some default allocation)
+    lenrw = 500000;
+    rw.reset(new snopt::doublereal[lenrw]);
+    leniw = 500000;
+    iw.reset(new snopt::integer[leniw]);
+    lencw = 500;
```

Comment:
It's cleaner to roll the factor of 8 into this instead of multiplying it out many times later.

```C++```
lencw = 8 * 500;
```
or

```C++```
lencw = 500;
lencw_bytes = 8 * 500;
```
I count 29 instances of `8*lencw`